WordPress/src/main/java/org/wordpress/android/ui/suggestion/adapters/SuggestionAdapter.java

```diff
@@ -112,41 +115,45 @@ public Filter getFilter() {
     private class SuggestionFilter extends Filter {
         @Override
         protected FilterResults performFiltering(CharSequence constraint) {
+            List<Suggestion> filteredSuggestions = getFilteredSuggestions(constraint);
             FilterResults results = new FilterResults();
+            results.values = filteredSuggestions;
+            results.count = filteredSuggestions.size();
+            return results;
+        }
 
+        @NonNull
+        private List<Suggestion> getFilteredSuggestions(CharSequence constraint) {
             if (mOrigSuggestionList == null) {
-                results.values = null;
-                results.count = 0;
+                return Collections.emptyList();
             } else if (constraint == null || constraint.length() == 0) {
-                results.values = mOrigSuggestionList;
-                results.count = mOrigSuggestionList.size();
+                return mOrigSuggestionList;
             } else {
-                List<Suggestion> nSuggestionList = new ArrayList<>();
-
+                List<Suggestion> filteredSuggestions = new ArrayList<>();
                 for (Suggestion suggestion : mOrigSuggestionList) {
                     String lowerCaseConstraint = constraint.toString().toLowerCase(Locale.getDefault());
-                    if (suggestion.getUserLogin().toLowerCase(Locale.ROOT).startsWith(lowerCaseConstraint)
-                        || suggestion.getDisplayName().toLowerCase(Locale.getDefault()).startsWith(lowerCaseConstraint)
-                        || suggestion.getDisplayName().toLowerCase(Locale.getDefault())
-                                     .contains(" " + lowerCaseConstraint)) {
-                        nSuggestionList.add(suggestion);
+                    boolean suggestionMatchesConstraint =
+                            suggestion.getUserLogin().toLowerCase(Locale.ROOT).startsWith(lowerCaseConstraint)
+                            || suggestion.getDisplayName().toLowerCase(Locale.getDefault())
+                                         .startsWith(lowerCaseConstraint)
+                            || suggestion.getDisplayName().toLowerCase(Locale.getDefault())
+                                         .contains(" " + lowerCaseConstraint);
+                    if (suggestionMatchesConstraint) {
```

Comment:
Almost suggested a unit test here, but then saw it was really just three intuitive cases, and I tested to make sure they all work fine. Looks good